### PR TITLE
#19 - Added actual services to service index response

### PR DESCRIPTION
**Pull request by olegmoz**
Part of issue #19 
Added Service interface to represent services that could present in service index. Added PackagePublish and PackageContent services

### CODE REVIEW

**Comment by paulodamaso** [comment]
@g4s8 PLease assign me

**Comment by 0crat** [comment]
This pull request [#54](https://github.com/artipie/nuget-adapter/issues/54) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/8be2bbff-614c-4334-8de7-6fd58747e0d8); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artpie/nuget/http/RouteService.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        this.base = base;
+        this.route = route;
+        this.stype = stype;
+    }
+
+    @Override
+    public String url() {
+        final String path = String.format("%s%s", this.base.getPath(), this.route.path());
+        final String file;
+        if (this.base.getQuery() == null) {
```
Comment (paulodamaso) [inline src/main/java/com/artpie/nuget/http/RouteService.java]: @olegmoz Please remove this `null` reference and use `Optional` here

File: src/test/java/com/artpie/nuget/http/NuGetServiceIndexTest.java [hunk 2]
```diff
@@ -72,7 +82,20 @@ void shouldGetIndex() {
             new AllOf<>(
                 Arrays.asList(
                     new RsHasStatus(RsStatus.OK),
-                    new RsHasBody(new IsValidServiceIndex())
+                    new RsHasBody(
+                        new IsValidServiceIndex(
+                            Matchers.contains(
```
Comment (paulodamaso) [inline src/test/java/com/artpie/nuget/http/NuGetServiceIndexTest.java]: @olegmoz Please don't use static matchers

**Comment by olegmoz** [comment]
@paulodamaso made changes by your comments, please review

**Comment by paulodamaso** [comment]
@g4s8 Please merge

**Comment by 0crat** [comment]
Job was finished in 42 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR

**Pull request merged**